### PR TITLE
Update halide mask and warp routine

### PR DESCRIPTION
**Pull request by antonysigma**
Hi @SteveDiamond , here is another round of Halide code update. This time, the following routines are updated:

 - `A_mask.cpp`: element-wise multiplication
 - `A_warp.cpp`: forward and backward perspective transform

I also rewrote the example Python code to reflect the changes.

Could you please review them for me?

### CODE REVIEW

File: proximal/examples/basic_denoise.py [hunk 1]
```diff
@@ -9,17 +9,30 @@
 from proximal.utils.utils import Impl
```
Comment (SteveDiamond) [inline proximal/examples/basic_denoise.py:9]: Why did you change this example?

File: proximal/utils/utils.py [hunk 2]
```diff
@@ -51,9 +51,12 @@ def _stack(arrays, axis=0):
 # Image utils
 ###############################################################################
 
-def get_test_image(WIDTH):
-    I = cv2.resize(ascent().astype(np.float32), (WIDTH, WIDTH))
-    I /= 255.
+def get_test_image(WIDTH, color=False):
+    if not color:
+        I = cv2.resize(ascent().astype(np.float32), (WIDTH, WIDTH))
```
Comment (SteveDiamond) [inline proximal/utils/utils.py:56]: S

**Comment by SteveDiamond** [comment]
It generally looks reasonable, I had a couple questions.

File: proximal/examples/basic_denoise.py (hunk 1 above)
Comment (antonysigma) [reply proximal/examples/basic_denoise.py]: My mistake. I am reverting it.

File: proximal/utils/utils.py (hunk 2 above)
Comment (antonysigma) [reply proximal/utils/utils.py:56]: You are correct. Thank you for spotting the error.

**Review by antonysigma — COMMENTED** [review]
Hi @SteveDiamond ,

Thank you for spotting the code errors. I have uploaded new changes to fix them.

Could you please review them for me?

Antony

**Comment by SteveDiamond** [comment]
Great, LGTM.
